tensorboard/webapp/app_routing/deep_link_provider.ts

```diff
@@ -20,11 +20,30 @@ import {State} from '../app_state';
 
 import {SerializableQueryParams} from './types';
 
+/**
+ * The `DeepLinkProvider` interface standardizes the relationship between the
+ * app state and the URL parameters.  It is expected that when either an update
+ * to the URL parameters should change app state, or, in the other direction,
+ * when an update to the state should cause a change in the URL parameters,
+ * that logic should be invoked through a `DeepLinkProvider`.
+ */
 @Injectable()
 export abstract class DeepLinkProvider {
+
+  /**
+   * Modifies the URL to match the state of the app.
+   * @param store The ngrx store containing the state of the app.
+   * @return An observable which will emit the query parameters matching the state.
+   */
   abstract serializeStateToQueryParams(
     store: Store<State>
   ): Observable<SerializableQueryParams>;
 
+  /**
+   * Given the query params, updates the ngrx state to match.
```

Comment:
Again, this method does not do the mutation; rather, it only returns a data structure, as per definition in the interface, based on current query parameter.